File: Mysql task/employee_management.py
import mysql.connector
from tabulate import tabulate

def get_connection():
    return mysql.connector.connect(
        host="localhost",
        user="root",
        password="password@123",
        database="company"
    )

def create_table(cursor):
    cursor.execute("""
        CREATE TABLE IF NOT EXISTS employees (
            EmpID VARCHAR(10) PRIMARY KEY,
            Name VARCHAR(50),
            Salary INT,
            Designation VARCHAR(50)
        )
    """)

def insert_initial_data(cursor, conn):
    employees = {
        "E101": {"Name": "Alice", "Salary": 55000, "Designation": "Developer"},
        "E102": {"Name": "Bob", "Salary": 60000, "Designation": "Manager"},
        "E103": {"Name": "Charlie", "Salary": 48000, "Designation": "Analyst"},
        "E104": {"Name": "David", "Salary": 70000, "Designation": "Team Lead"},
        "E105": {"Name": "Eva", "Salary": 50000, "Designation": "Tester"}
    }
    for emp_id, info in employees.items():
        cursor.execute(
            "INSERT IGNORE INTO employees (EmpID, Name, Salary, Designation) VALUES (%s, %s, %s, %s)",
            (emp_id, info["Name"], info["Salary"], info["Designation"])
        )
    conn.commit()

def add_employee(cursor, conn, emp_id, name, salary, designation):
    cursor.execute("INSERT INTO employees (EmpID, Name, Salary, Designation) VALUES (%s, %s, %s, %s)",
                   (emp_id, name, salary, designation))
    conn.commit()
    print("Employee added.")

def get_all_employees(cursor):
    cursor.execute("SELECT * FROM employees")
    rows = cursor.fetchall()
    headers = ["EmpID", "Name", "Salary", "Designation"]
    print("\nAll Employees:")
    print(tabulate(rows, headers=headers, tablefmt="pretty"))

def update_employee(cursor, conn, emp_id, field, new_value):
    if field not in ["Name", "Salary", "Designation"]:
        print("Invalid field.")
        return
    query = f"UPDATE employees SET {field} = %s WHERE EmpID = %s"
    cursor.execute(query, (new_value, emp_id))
    conn.commit()
    print("Employee updated.")

def delete_employee(cursor, conn, emp_id):
    cursor.execute("DELETE FROM employees WHERE EmpID = %s", (emp_id,))
    conn.commit()
    print("Employee deleted.")

def main():
    conn = get_connection()
    cursor = conn.cursor()

    create_table(cursor)
    insert_initial_data(cursor, conn)

    while True:
        print("\n===== Employee Management Menu =====")
        print("1. View all employees")
        print("2. Add employee")
        print("3. Update employee")
        print("4. Delete employee")
        print("5. Exit")
        choice = input("Enter your choice: ")

        try:
            if choice == '1':
                get_all_employees(cursor)
            elif choice == '2':
                eid = input("Enter EmpID: ")
                name = input("Enter Name: ")
                salary = int(input("Enter Salary: "))
                designation = input("Enter Designation: ")
                add_employee(cursor, conn, eid, name, salary, designation)
            elif choice == '3':
                eid = input("Enter EmpID: ")
                field = input("Field to update (Name/Salary/Designation): ")
                new_value = input("Enter new value: ")
                if field == "Salary":
                    new_value = int(new_value)
                update_employee(cursor, conn, eid, field, new_value)
            elif choice == '4':
                eid = input("Enter EmpID to delete: ")
                delete_employee(cursor, conn, eid)
            elif choice == '5':
                print("Exiting...")
                break
            else:
                print("Invalid choice.")
        except Exception as e:
            print("Error:", e)

    cursor.close()
    conn.close()

if __name__ == "__main__":
    main()

File: Java tasks/EmployeeManagement.java
import java.sql.*;
import java.util.Scanner;

public class EmployeeManagement {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection(
            "jdbc:mysql://localhost:3306/company",
            "root",
            "password@123"
        );
    }

    private static void createTable(Connection conn) throws SQLException {
        String sql = "CREATE TABLE IF NOT EXISTS employees (" +
                     "EmpID VARCHAR(10) PRIMARY KEY," +
                     "Name VARCHAR(50)," +
                     "Salary INT," +
                     "Designation VARCHAR(50))";
        Statement stmt = conn.createStatement();
        stmt.execute(sql);
    }

    private static void insertInitialData(Connection conn) throws SQLException {
        String sql = "INSERT IGNORE INTO employees (EmpID, Name, Salary, Designation) VALUES (?, ?, ?, ?)";
        PreparedStatement ps = conn.prepareStatement(sql);

        Object[][] employees = {
            {"E101", "Alice", 55000, "Developer"},
            {"E102", "Bob", 60000, "Manager"},
            {"E103", "Charlie", 48000, "Analyst"},
            {"E104", "David", 70000, "Team Lead"},
            {"E105", "Eva", 50000, "Tester"}
        };

        for (Object[] emp : employees) {
            ps.setString(1, (String) emp[0]);
            ps.setString(2, (String) emp[1]);
            ps.setInt(3, (int) emp[2]);
            ps.setString(4, (String) emp[3]);
            ps.executeUpdate();
        }
    }

    private static void getAllEmployees(Connection conn) throws SQLException {
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery("SELECT * FROM employees");

        System.out.printf("\n%-10s %-15s %-10s %-15s\n", "EmpID", "Name", "Salary", "Designation");
        System.out.println("------------------------------------------------------------");
        while (rs.next()) {
            System.out.printf("%-10s %-15s %-10d %-15s\n",
                rs.getString("EmpID"),
                rs.getString("Name"),
                rs.getInt("Salary"),
                rs.getString("Designation"));
        }
    }

    private static void addEmployee(Connection conn, Scanner scanner) throws SQLException {
        System.out.print("Enter EmpID: ");
        String empId = scanner.next();
        scanner.nextLine(); // consume newline
        System.out.print("Enter Name: ");
        String name = scanner.nextLine();
        System.out.print("Enter Salary: ");
        int salary = scanner.nextInt();
        scanner.nextLine(); // consume newline
        System.out.print("Enter Designation: ");
        String designation = scanner.nextLine();

        String sql = "INSERT INTO employees (EmpID, Name, Salary, Designation) VALUES (?, ?, ?, ?)";
        PreparedStatement ps = conn.prepareStatement(sql);
        ps.setString(1, empId);
        ps.setString(2, name);
        ps.setInt(3, salary);
        ps.setString(4, designation);
        ps.executeUpdate();
        System.out.println("Employee added.");
    }

    private static void updateEmployee(Connection conn, Scanner scanner) throws SQLException {
        System.out.print("Enter EmpID: ");
        String empId = scanner.next();

        System.out.print("Field to update (Name/Salary/Designation): ");
        String field = scanner.next();

        if (!field.equals("Name") && !field.equals("Salary") && !field.equals("Designation")) {
            System.out.println("Invalid field.");
            return;
        }

        System.out.print("Enter new value: ");
        scanner.nextLine();
        String newValue = scanner.nextLine();

        String sql = "UPDATE employees SET " + field + " = ? WHERE EmpID = ?";
        PreparedStatement ps = conn.prepareStatement(sql);
        if (field.equals("Salary")) {
            ps.setInt(1, Integer.parseInt(newValue));
        } else {
            ps.setString(1, newValue);
        }
        ps.setString(2, empId);
        int rows = ps.executeUpdate();
        if (rows > 0) {
            System.out.println("Employee updated.");
        } else {
            System.out.println("Employee not found.");
        }
    }

    private static void deleteEmployee(Connection conn, Scanner scanner) throws SQLException {
        System.out.print("Enter EmpID to delete: ");
        String empId = scanner.next();

        String sql = "DELETE FROM employees WHERE EmpID = ?";
        PreparedStatement ps = conn.prepareStatement(sql);
        ps.setString(1, empId);
        int rows = ps.executeUpdate();
        if (rows > 0) {
            System.out.println("Employee deleted.");
        } else {
            System.out.println("Employee not found.");
        }
    }

    public static void main(String[] args) {
        try (Connection conn = getConnection(); Scanner scanner = new Scanner(System.in)) {
            createTable(conn);
            insertInitialData(conn);

            while (true) {
                System.out.println("\n===== Employee Management Menu =====");
                System.out.println("1. View all employees");
                System.out.println("2. Add employee");
                System.out.println("3. Update employee");
                System.out.println("4. Delete employee");
                System.out.println("5. Exit");
                System.out.print("Enter your choice: ");
                String choice = scanner.next();

                switch (choice) {
                    case "1":
                        getAllEmployees(conn);
                        break;
                    case "2":
                        addEmployee(conn, scanner);
                        break;
                    case "3":
                        updateEmployee(conn, scanner);
                        break;
                    case "4":
                        deleteEmployee(conn, scanner);
                        break;
                    case "5":
                        System.out.println("Exiting...");
                        return;
                    default:
                        System.out.println("Invalid choice.");
                }
            }

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}
